### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by UQpfister**
This project tries to classify Alzheimer’s disease images of the ADNI brain dataset. The Siamese network is used to solve the problem. As a convolution neural network, a VGG16 network is used for the left and right sides of the Siamese network. In addition, as the loss function, contrastive loss is used. This project achieves an accuracy of 0.6 and has a test loop that also allows the plotting of test image pairs with the prediction and actual label.

### CODE REVIEW

**Comment by nathasha-naranpanawa** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme: Good
- Clear flow of information
- Algortihm and training choices explained well
- Relevant plots provided

Commit messages: OK

Code: 
- design is ok
- sufficiently commented
- siamese seem to be implemented correctly
- lots of repeated code, could use more functions

Functionality/Performance: 
- **only performing similarity detection between pairs of images, no classification**
- siamese network by itself only achieves 60% accuracy on the test set

General comments:
- **algorithm solves the respective problem only half-way**
- you need to have added a classifier that can classify an individual image as AD or NC with their features extracted by the trained siamese network

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem partially, no classifier for single images -3
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages mostly could be better -1
Progressive commits used
## Documentation
ReadMe acceptable/good
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description minimal -1
